src/backend/distributed/transaction/backend_data.c

```diff
@@ -400,6 +406,8 @@ InitializeBackendData(void)
 
 	Assert(MyBackendData);
 
+	LockBackendSharedMemory(LW_SHARED);
```

Comment:
This should be exclusive, otherwise it seems to falsify the comments above `LockBackendSharedMemory()`?